var/spack/repos/builtin/packages/rivet/package.py

```diff
@@ -155,6 +156,8 @@ def setup_build_environment(self, env):
         env.unset('PYTHONHOME')
         fjcontrib_home = self.spec['fjcontrib'].prefix
         env.prepend_path('LD_LIBRARY_PATH', fjcontrib_home.lib)
+        # Fix testNaN.sh and three other tests which could not find libHepMC.so.4
+        env.append_path('LD_LIBRARY_PATH', self.spec['hepmc'].prefix.lib)
```

Comment:
This looks sketchy. Can you try changing the hepmc dependency types instead? To:
```
   depends_on('hepmc',  type=('build', 'link', 'run'), when='hepmc=2')
   depends_on('hepmc3', type=('build', 'link', 'run'), when='hepmc=3')
```
 I think actually the default of `('build', 'link')` should be ok here too.

Reply:
I'm never sure about dependencies for tests though (). In any case you should `prepend_path` instead of `append_path`.

Reply:
@vvolkl :+1: :100: :1st_place_medal: Thanks! This did it! I do not know what the effect of 'run' is in detail, kept it just to be safe:
```diff
-    depends_on('hepmc',  type=('build', 'run'), when='hepmc=2')
-    depends_on('hepmc3', type=('build', 'run'), when='hepmc=3')
+    depends_on('hepmc',  type=('build', 'link', 'run'), when='hepmc=2')
+    depends_on('hepmc3', type=('build', 'link', 'run'), when='hepmc=3')
```
Pushed and ready for review!

Reply:
Sure. `type=('run')` will set up the dependency at runtime, so for example when you do `spack load rivet`. This is typically needed for python packages, to have the PYTHONPATH set up at runtime, and some packages may call the executables of other packages at runtime, but for hepmc I doubt it is needed. But I can clean this up in a future PR.